js/xt.js

```diff
@@ -0,0 +1,1072 @@
+const Exchange = require ('./base/Exchange');
+const { TICK_SIZE } = require ('./base/functions/number');
+const Precise = require ('./base/Precise');
+const { ExchangeError } = require ('./base/errors');
+// ^^^ BadSymbol, BadRequest, OnMaintenance, InvalidOrder, ArgumentsRequired AccountSuspended, PermissionDenied, RateLimitExceeded, ExchangeNotAvailable, OrderNotFound, InsufficientFunds, AuthenticationError
+
+module.exports = class xt extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'xt',
+            'name': 'XT',
+            'countries': [ 'SC' ], // Seychelles
+            // 10 requests per second => 1000ms / 10 = 100 (All other)
+            // 3 requests per second => 1000ms / 3 = 333.333 (get assets -> fetchMarkets & fetchCurrencies)
+            // 1000 times per minute for each single IP -> Otherwise account locked for 10min
+
+            'rateLimit': 100, // TODO: Is rate limit right? https://doc.xt.com/#documentationlimitRules
+            'version': '1',
+            'pro': false,
+            'has': {
+                'CORS': false,
+                'spot': true,
+                'margin': false,
+                'swap': true,
+                'future': true,
+                'option': undefined,
+                'fetchCurrencies': true,
+                'fetchMarkets': true,
+                'fetchSpotMarkets': true,
+                'fetchSwapAndFutureMarkets': true,
+                'fetchOHLCV': true,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchOrderbook': true,
+                'fetchTrades': true,
+            },
+            'precisionMode': TICK_SIZE,
+            'urls': {
+                'logo': '', // TODO: Add logo
+                'api': {
+                    'spot': {
+                        'public': 'https://api.xt.com',
+                        'private': 'https://api.xt.com',
+                    },
+                    'spot2': {
```

Comment:
what's the difference between spot and spot2?

Reply:
spot2 is the spare spot API basepoint